x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IndexLifecycleRunner.java

```diff
@@ -408,7 +409,7 @@ ClusterState moveClusterStateToFailedStep(ClusterState currentState, String inde
         ClusterState newState;
         IndexMetaData indexMetaData = currentState.metaData().index(index);
         if (indexMetaData == null) {
-            throw new IllegalArgumentException("index [" + index + "] does not exist");
+            throw new IndexNotFoundException(index);
```

Comment:
This has the implication that some APIs (ie. `ilm/_retry`) will return `404` instead of `400` which is a breaking change. I don't think we should make this change as part of this PR. Thoughts @dakrone ?

Reply:
We can stick with the existing exception, we can discuss making this change as a separate PR at a later time, thanks for catching this.

Reply:
Pushed the reverts for the exception change